include/common/enum_errcode.h

```diff
@@ -57,20 +57,45 @@ static inline constexpr auto WasmPhaseStr = []() constexpr {
     (defined(_MSVC_LANG) && _MSVC_LANG > 201402L)
 namespace WasmEdge {
 
+/// Error category C++ enumeration class.
+enum class ErrCategory : uint8_t {
+#define UseErrCategory
+#define Line(NAME, VALUE) NAME = VALUE,
+#include "enum.inc"
+#undef Line
+#undef UseErrCategory
+};
+
+} // namespace WasmEdge
+#endif
+
+/// Error category C enumeration.
+enum WasmEdge_ErrCategory {
+#define UseErrCategory
+#define Line(NAME, VALUE) WasmEdge_ErrCategory_##NAME = VALUE,
+#include "enum.inc"
+#undef Line
+#undef UseErrCategory
+};
+
+#if (defined(__cplusplus) && __cplusplus > 201402L) ||                         \
+    (defined(_MSVC_LANG) && _MSVC_LANG > 201402L)
+namespace WasmEdge {
+
 /// Error code C++ enumeration class.
-enum class ErrCode : uint8_t {
+enum class ErrEnum : uint32_t {
```

Comment:
We should use `ErrCodeValue`. The `ErrEnum` is pretty wierd.

Reply:
Maybe we should move this enum class into `class ErrCode` as its member enum directly.